src/utils/helpers/GraphHelpers.ts

```diff
@@ -75,6 +75,16 @@ export function highlight(clickedNode: DependencyNode) {
     });
 }
 
+function getDefaultScaleValue(container: NodeSelection, dimension: ReturnType<typeof findGroupBackgroundDimension>) {
+    if (!dimension) {
+        return 1;
+    }
+    const width = Number(container.attr('width'));
+    const height = Number(container.attr('height'));
+
+    return Math.min(1.3, 0.9 / Math.max(dimension.width / width, dimension.height / height));
```

Comment:
What is the reasoning behind this scale calculation? For me it's completely opaque.

Reply:
It chooses either biggest possible scale value(1.3 zoom value is a perfect to show focus on certain nodes and still allow user to see a bit wider context) or 90% of selected dimension to absolute(not dependent of zoom value) client width/height ratio - whatever is bigger.

I haven't thought about rewriting this before and I have no idea if the idea of deriving this value could be improved but I agree that it has some magic numbers and it isn't clear at all how the scale value is being derived.

I will try to extract some of the calculations to consts so it gets more readable